crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -469,6 +469,10 @@ define_tedge_config! {
             /// Set of SmartREST template IDs the device should subscribe to
             #[tedge_config(example = "templateId1,templateId2", default(function = "TemplatesSet::default"))]
             templates: TemplatesSet,
+
+            /// Switch using 501-503 (without operation ID) or 504-506 (with operation ID) SmartREST messages for operation status update
+            #[tedge_config(example = "true", default(value = true))]
+            with_operation_id: bool,
```

Comment:
What will be the behavior if operation ids are enabled on the device but not available on the c8y instance? Would things work? If not how the user will be notified? By error messages or stuck operations?

Reply:
> What will be the behavior if operation ids are enabled on the device but not available on the c8y instance? Would things work?

No, it won't work (that's we want to improve with `auto` option)

> If not how the user will be notified? By error messages or stuck operations?

I can't really test it since all tenants available for us is already 10.18 or above. User will definitely receive an error message on `c8y/s/e` for unknown static template like below.  

```
[c8y/s/us] 900
[c8y/s/e] 40,900,No static template for this message id
```

In this case, operation will be executed and successful/failed in thin-edge, but the status will stay `PENDING` in c8y.

Reply:
> What's the plan to implement auto detection? Can we probe for errors on c8y/s/e when sending 504-506 messages and then fallback to the previous API?

Definitely it's one of the doable options. We have similar issue for Advanced Software Management (requirement: 10.14> and dedicated Microservice). Want to have a consistent solution for this and Advanced Software Management. Probably we need to discuss Device Management Team to choose the best solution.